src/main/java/edu/texas/social_computing/hospitals/Matching.java

```diff
@@ -167,6 +172,108 @@ public void validateStability(List<Resident> residents, HospitalTable hospitalTa
         }
     }
 
+    private String getPrefString(List<String> preferences) {
+        StringBuilder prefString = new StringBuilder();
+        for (String p: preferences) {
+            prefString.append(p).append(" ");
+        }
+        return prefString.toString().trim();
+    }
+
+    public void outputMatchingToCsv(String filePrefix, List<Resident> residents, ResidentTable residentTable, List<Hospital> hospitals) {
+        List<String> resCSV = new ArrayList<>();
+        List<String> hospCSV = new ArrayList<>();
+        Charset utf8 = StandardCharsets.UTF_8;
+        // add headers
+        List<String> resHeaders = new ArrayList<>(Arrays.asList(
+                "residentId",
+                "residentAssignment",
+                "residentAssignmentLocation",
+                "residentAssignmentRank",
+                "residentInitialPreferences",
+                "residentFinalPreferences",
+                "hasPartner",
+                "isAnchorPartner",
+                "partnerId",
+                "partnerAssignment",
+                "partnerAssignmentLocation",
+                "partnerAssignmentRank",
+                "partnerInitialPreferences",
+                "partnerFinalPreferences"
+        ));
+        resCSV.add(String.join(", ", resHeaders));
+        List<String> hospHeaders = new ArrayList<>(Arrays.asList(
+                "hospitalId",
+                "hospitalLocation",
+                "capacity",
+                "assignmentCount",
+                "preferences",
+                "assignments",
+                "assignmentRanks"
+        ));
+        hospCSV.add(String.join(", ", hospHeaders));
+        // get all Residents
+        for (Resident resident : residents) {
+            List<String> resInitPrefs = resident.getInitialPreferences();
+            List<String> resPrefs = resident.getPreferences();
+            Hospital residentAssignment = getAssignedHospital(resident);
+            String partnerId = resident.getPartnerId();
+            Resident partner = residentTable.getResidentById(partnerId);
+            Hospital partnerAssignment = getAssignedHospital(partner);
+            String hasPartner = resident.hasPartner() ? "true" : "false";
```

Comment:
This should be a boolean. Then you can change all the hasPartner.equals("true") to hasPartner